code/game/verbs/suicide.dm

```diff
@@ -8,8 +8,12 @@
 	if(!canSuicide())
 		return
 	if(confirm == "Yes")
-		suiciding = 1
 		var/obj/item/held_item = get_active_hand()
+		if(istype(held_item, /obj/item/weapon/sord)) //I should really expand this so other items can easily prevent suicide as well...

```

Comment:
Ok, fine. Lemme see if i can figure out to make a "can'tsuicide" var for weapons.